README.md

```diff
@@ -28,311 +27,429 @@ def deps do
-]
-
-Kadena.Pact.ExecCommand.new(args)
+alias Kadena.Pact
+
+network_id = :testnet04
 
+Pact.ExecCommand.new() |> Pact.ExecCommand.set_network(network_id)
 ```
 
-#### NetworkID
+#### Code
+Executable PACT code.
 
-There are three options allowed to set a NetworkID:
+```elixir
+alias Kadena.Pact
 
-- `:testnet04`
-- `:mainnet01`
-- `:development`
+code = "(+ 1 2)"
 
-#### Code
+Pact.ExecCommand.new() |> Pact.ExecCommand.set_code(code)
+```
+
+#### Metadata
+Public metadata for Chainweb.
 
-A String value that represents the Pact code to execute in the `Execution Command`.
+```elixir
+alias Kadena.Pact
+
+metadata = Kadena.Types.MetaData.new(
+    creation_time: 1_667_249_173,
+    ttl: 28_800,
+    gas_limit: 1000,
+    gas_price: 0.01,
+    sender: "k:37e60c00779cacaef1f0a8697387a5945ef3cb82963980db486dc26ec5f424d9",
+    chain_id: "0"
+  )
+
+Pact.ExecCommand.new() |> Pact.ExecCommand.set_metadata(metadata)
+```
 
 #### Nonce
+An arbitrary user-supplied value. Defaults to current timestamp.
 
-A String value to ensure unique hash. You can use current timestamp.
+```elixir
+alias Kadena.Pact
 
-#### Step
+nonce = "2023-01-01 00:00:00.000000 UTC"
 
-The step is an Integer of the multi-step transaction.
+Pact.ExecCommand.new() |> Pact.ExecCommand.set_nonce(data)
+```
 
-#### Proof
+#### EnvData
+Environment transaction data.
 
-A String SPV proof, required for cross-chain transfer.
+```elixir
+alias Kadena.Pact
 
-#### Rollback
+env_data = %{
+    accounts_admin_keyset: [
+      "ba54b224d1924dd98403f5c751abdd10de6cd81b0121800bf7bdbdcfaec7388d"
+    ]
+  }
 
-A Boolean that indicates if the continuation is:
+Pact.ExecCommand.new() |> Pact.ExecCommand.set_data(data)
+```
 
-- rollback `true`
-- cancel `false`
+#### KeyPairs
+List of KeyPairs for signing.
 
-#### PactId
+```elixir
+alias Kadena.Pact
+alias Kadena.Cryptography.KeyPair
 
-Pact transaction id as a String.
+{:ok, keypair1} = KeyPair.generate()
+{:ok, keypair2} = KeyPair.generate()
 
-#### EnvData
+# add a list of keypairs
+Pact.ExecCommand.new() |> Pact.ExecCommand.add_keypairs([keypair1, keypair2])
+
+# add a single keypair
+Pact.ExecCommand.new() |> Pact.ExecCommand.add_keypair(keypair1)
+```
 
-A map must be provided to create an environment data, for example:
+#### Signers
+List of signers for detached signatures.
 
 ```elixir
-data = %{
-  accounts_admin_keyset: [
-    "ba54b224d1924dd98403f5c751abdd10de6cd81b0121800bf7bdbdcfaec7388d"
-  ]
-}
+alias Kadena.Pact
 
-Kadena.Types.EnvData.new(data)
+signer1 = Kadena.Types.Signer.new(pub_key: "37e60c00779cacaef1f0a8697387a5945ef3cb82963980db486dc26ec5f424d9")
+signer2 = Kadena.Types.Signer.new(pub_key: "8567032f1fe8b99c657338cd46480d0ee1a86985626b16374099d8d406e4d313")
+
+# add a list of signers
+Pact.ExecCommand.new() |> Pact.ExecCommand.add_signers([signer1, signer2])
+
+# add a single signer
+Pact.ExecCommand.new() |> Pact.ExecCommand.add_signer(signer1)
 ```
 
-#### MetaData
+#### Step (Continuation command)
```

Comment:
@juanhurtado10 looks like you missed using the continuation command for the examples in the Step, Proof, Rollback and PactTxHash sections